src/clients/base-evernode-client.js

```diff
@@ -77,6 +77,10 @@ class BaseEvernodeClient {
 
         await this.xrplApi.connect();
 
+        const validKeyPairAssignment = await this.xrplApi.isValidKeyForAddress(this.accKeyPair.publicKey, this.xrplAcc.address);
+        if (!validKeyPairAssignment)
+            throw 'Invalid client';
```

Comment:
Let's not do this check in the connect(). Let's simply expose the validation check via the XrplAccount class so the developer can verify it if needed.